src/App.js

```diff
@@ -26,15 +26,17 @@ class App extends Component {
   }
 
   render() {
-    const componentPage = urlToComponentPageMap[window.location.pathname];
+    const componentUrl =
+      process.env.PUBLIC_URL ? window.location.pathname.split(process.env.PUBLIC_URL)[1] : window.location.pathname;
```

Comment:
I think this should switch off of `process.env.NODE_ENV`? Also this is a good place to destructure as variables:

```
const {PUBLIC_URL, NODE_ENV} = process.env;
```

Reply:
Either way, `PUBLIC_URL` needs to be set with the expected URL to use; what's the upshot of conditionalizing based on two variables rather than one?

Reply:
Oh no, I wasn't saying using 2 conditionals. I was thinking the condition is based on whether or not its prod/dev, which NODE_ENV gives us.

Reply:
I think it's a good idea to use the conditional based on NODE_ENV, since the PUBLIC_URL being different is more of a byproduct of the condition than the actual condition.

Reply:
OK, I didn't realize at first that `PUBLIC_URL` is set either way and is based on that condition.